File: src/server/python/serde.py
import os
import json
try:
    import msgpack
    MSGPACK_AVAILABLE = True
except ImportError:
    MSGPACK_AVAILABLE = False

class Serializer:
    def __init__(self):
        self.format = os.getenv('SERDE', 'JSON').upper()

        if self.format == 'MSGPACK':
            if not MSGPACK_AVAILABLE:
                raise ImportError("MessagePack não disponível. Instale msgpack-python ou use SERDE=JSON")
            self.serialize = self._serialize_msgpack
            self.deserialize = self._deserialize_msgpack
        else:
            self.serialize = self._serialize_json
            self.deserialize = self._deserialize_json

    def _serialize_json(self, data):
        return json.dumps(data, ensure_ascii=False).encode('utf-8')

    def _deserialize_json(self, data):
        return json.loads(data.decode('utf-8'))

    def _serialize_msgpack(self, data):
        return msgpack.packb(data)

    def _deserialize_msgpack(self, data):
        return msgpack.unpackb(data, raw=False)

# Instância global
serializer = Serializer()

File: src/client/node/serde.js
const msgpack = require('@msgpack/msgpack');

class Serializer {
    constructor() {
        this.format = process.env.SERDE || 'JSON';

        if (this.format.toUpperCase() === 'MSGPACK') {
            this.serialize = this._serializeMsgpack;
            this.deserialize = this._deserializeMsgpack;
        } else {
            this.serialize = this._serializeJson;
            this.deserialize = this._deserializeJson;
        }
    }

    _serializeJson(data) {
        return Buffer.from(JSON.stringify(data), 'utf-8');
    }

    _deserializeJson(data) {
        return JSON.parse(data.toString('utf-8'));
    }

    _serializeMsgpack(data) {
        return msgpack.encode(data);
    }

    _deserializeMsgpack(data) {
        return msgpack.decode(data);
    }
}

module.exports = new Serializer();
